packages/wallets/infinitywallet/src/adapter.ts

```diff
@@ -0,0 +1,242 @@
+
+    private _connecting: boolean;
+    private _wallet: InfinityWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(config: InfinityWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (window.solana?.isInfinityWallet) {
+                    this._readyState = WalletReadyState.Installed;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+    get url(): string {
+        openInfinityWallet(window.location.href)
```

Comment:
Please remove this. Getters should have no side effects, _especially_ side effects that open a window or application.